### ISSUE TITLE
xSQLServerSetup: Installation hangs without any error when using UNC path

### ISSUE THREAD

**Issue by johlju** [PowerShell/xSQLServer#183]
Installation hangs without any error. This happens when using UNC path in the `SourcePath` parameter and at the same time is not using SourceCredential.
After some debugging, it seems that `Start-Process` hangs when calling setup.exe and does not return.

**This will be solved when PR #123 is merged.**

I think the culprit is security. ;) This one that popup when I ran the command 
```
Start-Process -FilePath \\fs.company.local\images\SQL2016RTM-ENT\setup.exe
    -ArgumentList '/IAcceptSQLServerLicenseTerms="True" /Action="Install" /InstanceName="SQL2016" /UpdateEnabled="False" /Features="IS" /BrowserSvcStartupType="Automatic"'
```

![image](https://cloud.githubusercontent.com/assets/7189721/20031866/d44a7728-a37e-11e6-94b5-ccd28999b220.png)

You can get around this by adding the domain to the trusted zone Local Intranet. But that is only for the logged on user. So won't do it for DSC.
![image](https://cloud.githubusercontent.com/assets/7189721/20031870/e7d87bfa-a37e-11e6-9071-56d4272bd270.png)

Tried changing the local group policy to allow .exe. But that did not work. _Did not try setting this in the domain, since it opens up to much._
```
Local Computer Policy/User Configuration/Administrative Templates/Windows Components/Attachment Manager

Setting: Inclusion list for low file types 
Value: .exe
```

What did work was to run `Start-Process` with the `-NoNewWindow` parameter. It then use the Microsoft .NET Framework CasPol (CasPol.exe) to "override" this lite bump. Don't know any more than that. If someone know more what happens here, please enlighten me 😄 

```
PS C:\Windows\system32> Start-Process -FilePath \\fs.company.local\images\SQL2016RTM-ENT\setup.exe
    -ArgumentList '/IAcceptSQLServerLicenseTerms="True" /Action="Install" /InstanceName="SQL2016" /UpdateEnabled="False" /Features="IS" /BrowserSvcStartupType="Automatic"' -NoNewWindow

Microsoft (R) SQL Server 2016 13.00.1601.05
Copyright (c) 2016 Microsoft Corporation.  All rights reserved.

Microsoft .NET Framework CasPol 4.0.30319.33440
for Microsoft .NET Framework version 4.0.30319.33440
Copyright (C) Microsoft Corporation.  All rights reserved.

WARNING: The .NET Framework does not apply CAS policy by default. Any settings shown or modified by CasPol will only
affect applications that opt into using CAS policy.

Please see http://go.microsoft.com/fwlink/?LinkId=131738 for more information.


Success
Microsoft .NET Framework CasPol 4.0.30319.33440
for Microsoft .NET Framework version 4.0.30319.33440
Copyright (C) Microsoft Corporation.  All rights reserved.

WARNING: The .NET Framework does not apply CAS policy by default. Any settings shown or modified by CasPol will only
affect applications that opt into using CAS policy.

Please see http://go.microsoft.com/fwlink/?LinkId=131738 for more information.


Success
SQL Server 2016 transmits information about your installation experience, as well as other usage and performance data, t
o Microsoft to help improve the product. To learn more about SQL Server 2016 data processing and privacy controls, please see the Privacy Statement.
```

**Comment by johlju** [comment]
Found this that explains more about CasPol.

[Caspol.exe (Code Access Security Policy Tool)](https://msdn.microsoft.com/en-us/library/cb6t8dtz%28v=vs.110%29.aspx)
[Using CasPol to Fully Trust a Share](https://blogs.msdn.microsoft.com/shawnfa/2004/12/30/using-caspol-to-fully-trust-a-share/)

This would trust everything under the 'foldershare'. 
CasPol.exe -polchgprompt off -machine -addgroup 1.2 -url \\fs.company.local\foldershare\* FullTrust

So then Start-Process must call CasPol to trust the path in the `-Path` parameter when the parameter `-NoNewWindow' is used.

**Comment by johlju** [comment]
For reference. This error occurs when not using the parameter `SourceCredential`.

**Comment by johlju** [comment]
In PR #578 there is a new timeout parameter (`SetupProcessTimeout`) which is meant to partly resolve this by at lest throwing an error if the setup is not finished within 2 hours (default value).

**Comment by johlju** [comment]
With PR #578 I cannot reproduce this one. So I think this get's solved in PR #578 thanks to it now starts the process using Setup-Process.
